src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -66,8 +72,9 @@ public SpanContext Extract(IHeadersCollection headers)
 
             var parentId = ParseUInt64(headers, HttpHeaderNames.ParentId);
             var samplingPriority = ParseEnum<SamplingPriority>(headers, HttpHeaderNames.SamplingPriority);
+            var origin = ParseString(headers, HttpHeaderNames.Origin);
 
-            return new SpanContext(traceId, parentId, samplingPriority);
+            return new SpanContext(traceId, parentId, samplingPriority, origin: origin);
```

Comment:
Can you also update the file `SpanContextPropagatorHelpers` with this logic? That helper class allows us to read/write the `System.Net.Http.Headers.HttpHeaders` class without statically referencing it, which is used in `Datadog.Trace.ClrProfiler.Integrations.HttpMessageHandlerIntegration` and `Datadog.Trace.ClrProfiler.Integrations.AspNetWebApi2Integration`

Reply:
Sure, I'll update it, didn't know about the helpers...

Reply:
Done.